project/npda/general_functions/rcpch_nhs_organisations.py

```diff
@@ -131,116 +135,102 @@ def get_all_nhs_organisations_affiliated_with_paediatric_diabetes_unit() -> (
-#       "publication_date": "2022-12-08"
-#     },
-#     "london_borough": null,
-#     "country": {
-#       "boundary_identifier": "E92000001",
-#       "name": "England"
-#     }
-#   },
-#   {
-#     "ods_code": "RCF22",
-#     "name": "AIREDALE GENERAL HOSPITAL",
-#     "website": "https://www.airedaletrust.nhs.uk/",
-#     "address1": "SKIPTON ROAD",
-#     "address2": "STEETON",
-#     "address3": "",
-#     "telephone": "",
-#     "city": "KEIGHLEY",
-#     "county": "WEST YORKSHIRE",
-#     "latitude": 53.8979454,
-#     "longitude": -1.962710142,
-#     "postcode": "BD20 6TD",
-#     "geocode_coordinates": "SRID=27700;POINT (-1.962710142 53.8979454)",
-#     "active": true,
-#     "published_at": null,
-#     "paediatric_diabetes_unit": {
-#       "pz_code": "PZ047"
-#     },
-#     "trust": {
-#       "ods_code": "RCF",
-#       "name": "AIREDALE NHS FOUNDATION TRUST",
-#       "address_line_1": "AIREDALE GENERAL HOSPITAL",
-#       "address_line_2": "SKIPTON ROAD",
-#       "town": "KEIGHLEY",
-#       "postcode": "BD20 6TD",
-#       "country": "ENGLAND",
-#       "telephone": null,
-#       "website": null,
-#       "active": true,
-#       "published_at": null
-#     },
-#     "local_health_board": null,
-#     "integrated_care_board": {
-#       "boundary_identifier": "E54000054",
-#       "name": "NHS West Yorkshire Integrated Care Board",
-#       "ods_code": "QWO"
-#     },
-#     "nhs_england_region": {
-#       "region_code": "Y63",
-#       "publication_date": "2022-07-30",
-#       "boundary_identifier": "E40000012",
-#       "name": "North East and Yorkshire"
-#     },
-#     "openuk_network": {
-#       "name": "Yorkshire Paediatric Neurology Network",
-#       "boundary_identifier": "YPEN",
-#       "country": "England",
-#       "publication_date": "2022-12-08"
-#     },
-#     "london_borough": null,
-#     "country": {
-#       "boundary_identifier": "E92000001",
-#       "name": "England"
-#     }
-#   },
+def get_all_pz_codes_with_their_trust_and_primary_organisation(
+    seed=False,
+) -> List[Tuple[str, str]]:
+    """
+    This function returns all NHS organisations from the RCPCH dataset that are affiliated with a paediatric diabetes unit.
+    Accepts a seed parameter - if True, it will seed the database with the data.
+    If False, it will return the data as a list of tuples.
+
+    If an error occurs while fetching the data, it returns a list with a single tuple containing:
+        [("999", "An error occurred while fetching NHS organisations.")].
+
+    Returns:
+        List[Tuple[str, str]]: A list of tuples containing the ODS code and name of NHS organisations.
+    """
+    url = f"{settings.RCPCH_NHS_ORGANISATIONS_API_URL}/paediatric_diabetes_units/trust"
+    ERROR_RESPONSE = [("999", "An error occurred while fetching NHS organisations.")]
+
+    try:
+        response = requests.get(url=url, timeout=10)  # times out after 10 seconds
+        response.raise_for_status()
+
+        if seed:
```

Comment:
I think this function is only called with `seed=True`. Should we remove the argument entirely and the code below this guard?